### DISCUSSION TITLE
RFC: Image component with next export

### DISCUSSION THREAD

**Discussion by samrobbins85** [vercel/next.js#20659] [Ideas]
# If you found this issue, please go to #19065 and vote ( +1 / :arrow_up: ) there! (The maintainer are using the votes, to prioritize work :smiley: )

## Goals

- A user of a static hosting platform who uses `next export` will get a warning rather than an error when using `next/image` and the default loader
- The layout of the image doesn't change with `next export`

## Background

When using `next export` and the default image loader you will get an error. This is because the loader works on the server in Vercel, and so can't be implemented in a static site. There are a few situations where this wouldn't be the desirable outcome
1. Forking a repository you want to host yourself. If the site is set up with `next/image` you wouldn't be able to use it without changes. This could somewhat hamper adoption of Next.js as people could just give up as it doesn't work.
2. Moving from Vercel to another platform. If there was a reason you needed to do this in a hurry, it could be a lot of work to go through and replace the image component in all locations
3. Hosting the same project on multiple platforms. This one is more niche, but could happen that someone wants double hosting, and the image component would prevent this working.

## Implementation

There are three different solutions I can think of, I don't know which are possible, but they're ordered with 1 being the best in my opinion and 3 being the worst, but multiple solutions could be implemented.

1. `next export` automatically implements the `<Image />` component as the base `img` (the only option that satisfies the 3rd point, although it is niche so may not be something needed to think about)
2. An additional loader is provided, `static`, which just passes through the image URL
3. A [codemod](https://nextjs.org/docs/advanced-features/codemods) is provided which transforms `<Image/>` to `img`

**Comment by drothmaler** [comment]
Seems to be a duplicate of #19065

**Reply by samrobbins85** [reply]
Oops, so it is, thanks for the spot!

**Reply by samrobbins85** [reply]
Looks like you can't delete discussions on GitHub, so I think I've just got to keep this around unfortunately

**Reply by drothmaler** [reply]
But this duplicate might direct people in the right direction; so guys: 

If you found this issue, please **go to #19065 and vote ( 👍 / ⬆️  ) there!** (The maintainer are using the votes, to prioritize work 😃 )

**Reply by samrobbins85** [reply]
Good point, put it at the top of the post as well to make it more prominent
